applications/tari_merge_mining_proxy/src/main.rs

```diff
@@ -736,30 +806,59 @@ fn handle_connection(
         let json_bytes = get_json(&buffer[..]);
         if let Some(json) = json_bytes {
             let method = get_post_json_method(&json);
+            debug!(target: LOG_TARGET, "Handling POST Method: {}", method);
             let request_id = get_request_id(&json);
             default_response = format!(
                 "{} \"error\": {} \"code\": \"{}\", \"message\": \"{}\"{}, \"id\": {}, \"jsonrpc\": \"2.0\"{}",
                 "{", "{", 418, "Curl failed to brew the request", "}", request_id, "}"
             )
             .as_bytes()
             .to_vec();
-            match base_curl(json.len() as u64, &url, true, config) {
-                Ok(mut curl) => {
-                    match do_curl(&mut curl, &json) {
-                        // request_id
+
+            if transient.tari_difficulty.unwrap_or_default() < transient.monero_difficulty.unwrap_or_default() &&
```

Comment:
We should look at the achieved difficulty here.  Not comparing the mined difficulty as is here currently. 
This should be a simple if :
`if achieved_block_difficulty >= tari_target_diff
submit tari

if achieved_block_difficulty >= monero_target_diff
submit monero
`
We mine to the target, but we submit/check to the achieved.

Reply:
While I understand this, we can assume that submitting to monero would have succeeded when only submitting to tari. But what happens if it had to fail, wouldn't we be presenting an incorrect response to XMRig (which displays success and failure stats)? Unfortunately, if it succeeds to monero and fails to tari there is not much we could do about it in that case.

We also do need to revisit the achieved difficulty calculation as previously mentioned.

Reply:
What do you mean we need to revisit the achieved difficulty? 

`we can assume that submitting to monero would have succeeded when only submitting to tari`
I completely disagree with this statement. What if tari has a higher difficulty? Then we need to submit only to monero. 

We dont care about submitting an incorrect block to monero, or tari for that matter. We care about getting as much as possible coins for the miner. If we can get Tari, or Monero, or both, we need to claim. If we can submit a valid block for one, we need to do that asp. We cant wait till we have both.